socks5.py

```diff
@@ -52,24 +68,28 @@
         else:
             iftypes['cell'].append(iface)
 
+
     if iftypes['bridge']:
         iface = iftypes['bridge'][0]
-        print("Assuming proxy will be accessed over hotspot (%s) at %s" %
-              (iface.name, iface.addr.address))
+        initial_output = "Assuming proxy will be accessed over hotspot (%s) at %s\n" % (iface.name, iface.addr.address)
         PROXY_HOST = iface.addr.address
     elif iftypes['en']:
         iface = iftypes['en'][0]
-        print("Assuming proxy will be accessed over WiFi (%s) at %s" %
-              (iface.name, iface.addr.address))
+        initial_output += "Assuming proxy will be accessed over WiFi (%s) at %s\n" % (iface.name, iface.addr.address)
         PROXY_HOST = iface.addr.address
     else:
-        print('Warning: could not get WiFi address; assuming %s' % PROXY_HOST)
+        initial_output += 'Warning: could not get WiFi address; assuming %s\n' % PROXY_HOST
 
     if iftypes['cell']:
-        iface = iftypes['cell'][0]
-        print("Will connect to servers over interface %s at %s" %
-              (iface.name, iface.addr.address))
-        CONNECT_HOST = iface.addr.address
+        iface_ipv4 = next((iface for iface in iftypes['cell'] if iface.addr.family == socket.AF_INET), None)
+        iface_ipv6 = next((iface for iface in iftypes['cell'] if iface.addr.family == socket.AF_INET6), None)
```

Comment:
How was this tested? There's an `if` condition that blocks non-`AF_INET` interfaces above (`TODO IPv6 support someday`, line 60). In particular, I am concerned about IPv6-only networks, which do now exist...

Reply:
The line above seems to be blocking non-AF_INET for the IPs you would connect to (eg: SOCKS address, PAC url).

I've tested this on both my iPhone with Pythonista (which has T-Mobile USA which runs native IPv6 and 464XLAT for IPv4-only hosts). Sites like www.ip6.me load with an IPv6 address, and ip4.me loads with an IPv4 address.

Reply:
I misunderstood the purpose of this blocking code because ... and because it was able to work without removing it, I was able to make it quite far in the development before encountering some limitations ... Now that I've commented out that line, it's exposed a few of other bugs which need to be addressed (and will be in the next few commits).

Thanks for your comments - Obviously you understand your code better than I did when I quickly skimmed it, and I would have gone in circles for hours trying to fix the bugs I'm running into without your comment pointing this out.